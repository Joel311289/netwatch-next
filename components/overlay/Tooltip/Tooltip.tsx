import { useState } from 'react';
import TooltipMUI from '@mui/material/Tooltip';
import { ElementChildren, ElementHTML } from '@/types';
import styles from '@/components/overlay/Tooltip/Tooltip.module.scss';

export type TooltipProps = typeof defaultProps &
  ElementHTML &
  ElementChildren & {
    text: string;
  };

const defaultProps = {
  name: 'Tooltip'
};

const Tooltip = ({ children, text }: TooltipProps) => {
  const [open, setOpen] = useState<boolean>(false);

  const handleTooltipChange = (enabled: boolean) => setOpen(enabled);

  return (
    <TooltipMUI
      title={text}
      classes={{ tooltip: styles.tooltipWrapper }}
      onClose={() => handleTooltipChange(false)}
      onOpen={() => handleTooltipChange(true)}
      onClick={() => handleTooltipChange(false)}
      open={open}
      PopperProps={{
        disablePortal: true
      }}
      arrow>
      <div className={styles.tooltipContent}>{children}</div>
    </TooltipMUI>
  );
};

Tooltip.defaultProps = defaultProps;

export default Tooltip;
