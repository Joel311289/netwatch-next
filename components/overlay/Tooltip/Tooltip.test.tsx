import { act, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Tooltip from '@/components/overlay/Tooltip/Tooltip';

describe('Tests Tooltip component', () => {
  test('renders component correctly', async () => {
    const { container } = render(
      <Tooltip text="Close">
        <button>X</button>
      </Tooltip>
    );

    expect(container).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    render(
      <Tooltip text="Close">
        <button>X</button>
      </Tooltip>
    );

    expect(screen.getByText('X')).toBeInTheDocument();
  });

  test('renders tooltip correctly', async () => {
    render(
      <Tooltip text="Close">
        <button>X</button>
      </Tooltip>
    );

    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(() => {
      userEvent.hover(screen.getByText('X'));
    });

    await waitFor(() => {
      expect(screen.getByText('Close')).toBeInTheDocument();
    });
  });
});
