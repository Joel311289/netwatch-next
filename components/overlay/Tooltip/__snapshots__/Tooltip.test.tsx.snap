// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Tooltip component renders component correctly 1`] = `
<div>
  <div
    aria-label="Close"
    class="tooltipContent"
    data-mui-internal-clone-element="true"
    role="tooltip"
  >
    <button>
      X
    </button>
  </div>
</div>
`;
